src/ccall/src/goat_game.rs

```diff
@@ -0,0 +1,82 @@
+use amethyst::ecs::prelude::{Component, DenseVecStorage};
+use amethyst::utils::scene::BasicScenePrefab;
+use amethyst::{
+    assets::{Handle, Loader, PrefabLoader, ProgressCounter, RonFormat},
+    core::{math::Vector3, Transform},
+    prelude::*,
+    renderer::{
+        camera::Camera,
+        rendy::mesh::{MeshBuilder, Position, PosNormTex},
+        types::{Mesh, MeshData},
+    },
+};
+
+use crate::goat::*;
+
+pub type MyPrefabData = BasicScenePrefab<Vec<PosNormTex>>;
+pub struct GoatGame;
+pub struct Model {}
+
+impl Component for Model {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl SimpleState for GoatGame {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        let StateData { world, .. } = data;
+
+        initialise_camera(world);
+        initialize_model(world);
+    }
+}
+
+// ToDo move camera to prefab and try hot-reloading
+fn initialise_camera(world: &mut World) {
+    let mut transform = Transform::default();
+    transform.set_translation_xyz(0.0, 0.0, -10.0);
+
+    world
+        .create_entity()
+        .with(Camera::standard_3d(100.0, 100.0))
```

Comment:
does this add a camera component?

Reply:
Yeah. If I put back in the skybox you'll see it but right now you can enjoy the coldness of the void.